src/CoreFoundation/NativeObject.cs

```diff
@@ -78,12 +77,10 @@ protected virtual void InitializeHandle (IntPtr handle)
 			this.handle = handle;
 		}
 
-		void Throw () => throw new ObjectDisposedException (GetType ().ToString ());
-
 		internal IntPtr GetCheckedHandle ()
 		{
 			if (handle == IntPtr.Zero)
-				Throw ();
+				throw new ObjectDisposedException (GetType ().ToString ());
```

Comment:
Yeah, but this prevents GetCheckedHandle from getting inlined.

Reply:
It's a common pattern (in dotnet/runtime) to have helper functions for throwing methods, so the containing methods have more chances to be inlined (and the cost of calling a new method is just in the scenario where the actual exception should be thrown).

Reply:
Thats actually the only reason they do it. I'm with @Therzok on this one.

Reply:
This is why things like [ThrowHelper](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Linq/src/System/Linq/ThrowHelper.cs#L10-L59) exist.

Reply:
heh, I already have a local `ThrowHelper` for something else ;-)

It's a performance vs size tradeoff (one much easier to measure than the other), which has not until recently been considered by dotnet/runtime (and they should now be considered as equals).

Reply:
Perf vs size has been considered by dotnet/runtime before, but possibly not in this specific scenario.

This looks like a generic enough method to be reused via a ThrowHelper though.

See speed vs size opt here: https://github.com/dotnet/runtime/tree/master/src/libraries/System.Linq/src/System/Linq

Reply:
I'd argue that is was a fix after the fact, not a consideration when the feature coding was done.

Now there's nothing wrong about putting performance first, it made a lot of sense for the original scenarios that netcore targeted. This is also true for mono. There's a lot of code, even in the Xamarin years, where performance alone (without size) was considered.

Expending the scenarios to include mobile means that code examples (like your linke) will become more common (at build time or thru the linker).

Ideally new code will consider both perf/size and not assume existing code did the same :)